packages/graphql-server/src/plugins/useRedwoodDirective.ts

```diff
@@ -91,79 +95,129 @@ export function hasDirective(info: GraphQLResolveInfo): boolean {
 }
 
 export function getDirectiveByName(
-  info: GraphQLResolveInfo,
-  name: string
+  fieldConfig: GraphQLFieldConfig<any, any, any>,
+  directiveName: string
 ): null | DirectiveNode {
-  try {
-    const { parentType, fieldName, schema } = info
-    const schemaType = schema.getType(parentType.name) as GraphQLObjectType
-    const field = schemaType.getFields()[fieldName]
-    const astNode = field.astNode
-    const associatedDirective = astNode?.directives?.find(
-      (directive) => directive.name.value === name
-    )
+  const associatedDirective = fieldConfig.astNode?.directives?.find(
+    (directive) => directive.name.value === directiveName
+  )
+  return associatedDirective ?? null
+}
 
-    return associatedDirective || null
-  } catch (error) {
-    console.error(error)
-    return null
-  }
+export function isPromise(value: any): value is Promise<unknown> {
+  return typeof value?.then === 'function'
 }
 
-export const useRedwoodDirective = (
+function wrapAffectedResolvers(
+  schema: GraphQLSchema,
   options: DirectivePluginOptions
-): Plugin<{
-  onResolverCalled: ValidatorDirectiveFunc | TransformerDirectiveFunc
-}> => {
-  return {
-    async onResolverCalled({ args, root, context, info }) {
-      const directiveNode = getDirectiveByName(info, options.name)
+): GraphQLSchema {
+  return mapSchema(schema, {
```

Comment:
See https://www.graphql-tools.com/docs/schema-directives#implementing-schema-directives for more context